### PR TITLE
Implement and test OpLog for update operations

### LINKED ISSUE #3556: Insert `i`/`u`/`d` documents into OpLog
### What should be done?

Insert i/u/d documents in the same format as MongoDB.
Do enough to make it barely compatible with Meteor apps.
Add issues and TODOs.

### Where?

https://github.com/FerretDB/FerretDB/tree/main/internal/backends/decorators/oplog

### Definition of Done

- integration/compatibility tests added/updated;
- spot refactorings done.

### LINKED ISSUE #3861: Add integration tests for OpLog i/u/d
### What should be done?

We should add integration tests for OpLog for simple operations: `insert` `update` and `delete`

Tests should `insert`, `update`, and `delete` some documents in some collection, then check that documents `local.oplog.rs` are what is expected. This should not be a compat test but a regular integration test.

Tests should not use tailable cursors.

They should pass for MongoDB and are expected to fail for FerretDB (for now).

### Where?

`integration/oplog` (new directory).

### Definition of Done

- integration tests added;
- they pass for MongoDB;
- they expectedly fail for FerretDB.

### PR DESCRIPTION
**Pull request by rumyantseva**
A working prototype: is in this draft: #3913

# Description

Closes #3861.
Closes #3556.

In this PR: the necessary changes to support oplog for update operations in `{$v: 1}` format.
In a separate PR: the changes related to replica set params necessary to connect meteor to FerretDB.

To see how it works, run with meteor's tic tac toe example.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/oplog/update_oplog_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	ns := fmt.Sprintf("%s.%s", coll.Database().Name(), coll.Name())
+	opts := options.FindOne().SetSort(bson.D{{"$natural", -1}})
+
+	err := local.CreateCollection(ctx, "oplog.rs", options.CreateCollection().SetCapped(true).SetSizeInBytes(536870912))
+	if err != nil {
+		require.Contains(t, err.Error(), "local.oplog.rs already exists")
+		err = nil
+	}
+
+	require.NoError(t, err)
```
Comment (henvic) [inline integration/oplog/update_oplog_test.go:47]: minor: have you considered the following instead?

```suggestion
	if err := local.CreateCollection(ctx, "oplog.rs", options.CreateCollection().SetCapped(true).SetSizeInBytes(536870912)); err != nil {
		require.Contains(t, err.Error(), "local.oplog.rs already exists")
	}
```

File: integration/oplog/update_oplog_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		update         bson.D
+		filter         bson.D
+		expectedDiffV1 *types.Document
+		expectedO2     *types.Document
```
Comment (henvic) [inline integration/oplog/update_oplog_test.go:53]: What is the meaning of O2?

Comment (rumyantseva) [reply integration/oplog/update_oplog_test.go:53]: o2 only exists for diff (o) with $v: 1 and contains the id of the oploged entry. At least, as far as I understand :)

**Review by henvic — DISMISSED** [review]

**Review by henvic — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]
:shipit:

**Pull request merged**